benchmarks/callbacks/benchmark_RegionTimer.cpp

```diff
@@ -0,0 +1,320 @@
+    explicit OverheadBenchmarkImpl(const ::benchmark::State&)
+    {
+        Kokkos::Tools::Experimental::set_push_region_callback(OverheadBenchmarkImpl::push);
+        Kokkos::Tools::Experimental::set_pop_region_callback (OverheadBenchmarkImpl::pop);
+    }
+
+    ~OverheadBenchmarkImpl()
+    {
+        Kokkos::Tools::Experimental::set_push_region_callback(nullptr);
+        Kokkos::Tools::Experimental::set_pop_region_callback (nullptr);
+    }
+
+    void run_push_pop() const
+    {
+        Kokkos::Tools::pushRegion(REGION_LABEL);
+        Kokkos::Tools::popRegion();
+    }
+
+    auto run_timer()
+    {
+        timer.start();
+        timer.stop();
+        return timer.template duration<Kokkos::utils::timer::nanoseconds>();
+    }
+
+private:
+    Kokkos::utils::timer::Timer<void> timer;
+};
+
+using SequenceOfRegionTimerListenerBenchmark = Fixture<SequenceOfRegionTimerListenerBenchmarkImpl>;
+
+BENCHMARK_DEFINE_F(SequenceOfRegionTimerListenerBenchmark, time_sequence_of_regions)(::benchmark::State& state) {
+    for (auto _ : state) {
+        this->impl->run(state);
+    }
+}
+BENCHMARK_REGISTER_F(SequenceOfRegionTimerListenerBenchmark, time_sequence_of_regions)
+    ->ArgNames({"repeat", "use-manager"})
+    ->Args({  0, true})->Args({  0, false})
+    ->Args({ 25, true})->Args({ 25, false})
+    ->Args({250, true})->Args({250, false});
+
+using ManagerBenchmark = Fixture<ManagerBenchmarkImpl>;
+
+BENCHMARK_DEFINE_F(ManagerBenchmark, time_sequence_of_regions)(::benchmark::State& state) {
+    for (auto _ : state) {
+        this->impl->run(state);
+    }
+}
+BENCHMARK_REGISTER_F(ManagerBenchmark, time_sequence_of_regions)
+    ->ArgName("repeat")
+    ->Arg(0)->Arg(25)->Arg(250);
+
+using OverheadBenchmark = Fixture<OverheadBenchmarkImpl>;
+
+BENCHMARK_DEFINE_F(OverheadBenchmark, push_pop)(::benchmark::State& state) {
+    for (auto _ : state) {
+        this->impl->run_push_pop();
+    }
+}
+BENCHMARK_REGISTER_F(OverheadBenchmark, push_pop);
+
+BENCHMARK_DEFINE_F(OverheadBenchmark, timer)(::benchmark::State& state)
+{
+    const auto num_iters = std::distance(state.begin(), state.end());
+
+    std::vector<double> reported_elapsed(num_iters, 0.);
+
+    size_t iter = 0;
+
+    for (auto _ : state) {
+        reported_elapsed.at(iter++) = this->impl->run_timer().count();
+    }
+    const double reported_elapsed_mean = std::accumulate(reported_elapsed.cbegin(), reported_elapsed.cend(), double(0.), std::plus{}) / num_iters;
+    const double reported_elapsed_std  = std::sqrt(std::accumulate(reported_elapsed.cbegin(), reported_elapsed.cend(), double(0.), [&](const double current, const double value){
+        return current + std::pow(value - reported_elapsed_mean, 2);
+    }) / num_iters);
+
+    state.counters["reported-mean"] = ::benchmark::Counter(reported_elapsed_mean, ::benchmark::Counter::Flags::kDefaults);
+    state.counters["reported-std" ] = ::benchmark::Counter(reported_elapsed_std,  ::benchmark::Counter::Flags::kDefaults);
```

Comment:
Just wondering if we want to keep this piece? Or was it just to double check once that the "relative error" criterion works? If we keep it, also wondering if the write into the vector also adds to the time that we're measuring.